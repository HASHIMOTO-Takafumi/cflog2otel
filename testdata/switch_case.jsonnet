local cel = std.native('cel');
local cel_switch = std.native('cel_switch');

{
  otel: {
    endpoint: 'http://localhost:4317/',
    gzip: true,
  },
  resource_attributes: [
    {
      key: 'service.name',
      value: 'Amazon CloudFront',
    },
    {
      key: 'aws.cloudfront.distribution_id',
      value: cel('cloudfront.distributionId'),
    },
  ],
  scope: {
    name: 'test',
  },
  metrics: [
    {
      name: 'http.server.5xx_requests',
      description: 'The number of HTTP requests with status code 5xx',
      type: 'Count',
      attributes: [
        {
          key: 'cloudfront.origin',
          value: cel_switch([
            {
              case: 'log.csUriStem.startsWith("/index.html")',
              value: 'S3',
            },
            {
              case: 'log.csUriStem == "/favicon.ico"',
              value: 'S3',
            },
            {
              default: 'app',
            },
          ]),
        },
        {
          key: 'http.status_code',
          value: cel('log.scStatusCategory'),
        },
      ],
    },
  ],
}
