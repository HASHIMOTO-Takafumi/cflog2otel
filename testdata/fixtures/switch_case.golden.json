{
  "Resource": [
    {
      "Key": "aws.cloudfront.distribution_id",
      "Value": {
        "Type": "STRING",
        "Value": "EMLARXS9EXAMPLE"
      }
    },
    {
      "Key": "service.name",
      "Value": {
        "Type": "STRING",
        "Value": "Amazon CloudFront"
      }
    }
  ],
  "ScopeMetrics": [
    {
      "Scope": {
        "Name": "test",
        "Version": "",
        "SchemaURL": ""
      },
      "Metrics": [
        {
          "Name": "http.server.5xx_requests",
          "Description": "The number of HTTP requests with status code 5xx",
          "Unit": "",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [
                  {
                    "Key": "cloudfront.origin",
                    "Value": {
                      "Type": "STRING",
                      "Value": "S3"
                    }
                  },
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "2xx"
                    }
                  }
                ],
                "StartTime": "2019-12-04T21:02:00Z",
                "Time": "2019-12-04T21:02:59.999999999Z",
                "Value": 3
              },
              {
                "Attributes": [
                  {
                    "Key": "cloudfront.origin",
                    "Value": {
                      "Type": "STRING",
                      "Value": "S3"
                    }
                  },
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-13T22:36:00Z",
                "Time": "2019-12-13T22:36:59.999999999Z",
                "Value": 1
              },
              {
                "Attributes": [
                  {
                    "Key": "cloudfront.origin",
                    "Value": {
                      "Type": "STRING",
                      "Value": "app"
                    }
                  },
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-13T22:36:00Z",
                "Time": "2019-12-13T22:36:59.999999999Z",
                "Value": 1
              },
              {
                "Attributes": [
                  {
                    "Key": "cloudfront.origin",
                    "Value": {
                      "Type": "STRING",
                      "Value": "app"
                    }
                  },
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-13T22:37:00Z",
                "Time": "2019-12-13T22:37:59.999999999Z",
                "Value": 1
              }
            ],
            "Temporality": "CumulativeTemporality",
            "IsMonotonic": true
          }
        }
      ]
    }
  ]
}